lib/matplotlib/gridspec.py

```diff
@@ -202,6 +202,31 @@ def get_grid_positions(self, fig, raw=False):
         fig_lefts, fig_rights = (left + cell_ws).reshape((-1, 2)).T
         return fig_bottoms, fig_tops, fig_lefts, fig_rights
 
+    @staticmethod
+    def _check_gridspec_exists(figure, nrows, ncols):
+        """
+        Check if the figure already has a gridspec with these dimensions,
+        or create a new one
+        """
+        gs = None
+        for ax in figure.get_axes():
+            if hasattr(ax, 'get_subplotspec'):
+                ggs = ax.get_subplotspec().get_gridspec()
+                if hasattr(ggs, 'get_topmost_subplotspec'):
+                    # This is needed for colorbar gridspec layouts.
+                    # This is probably OK becase this whole logic tree
+                    # is for when the user is doing simple things with the
+                    # add_subplot command.  For complicated layouts
+                    # like subgridspecs the proper gridspec is passed in...
+                    ggs = ggs.get_topmost_subplotspec().get_gridspec()
+                (nrow, ncol) = ggs.get_geometry()
+                if nrow == nrows and ncol == ncols:
+                    gs = ggs
```

Comment:
perhaps you can just `return ggs` here? (dunno if it matters whether you pick the first or the last match, hopefully there's only one anyways...) and then you don't need to have `gs = None` at the top, yada yada.

Reply:
Done.